sktime/forecasting/compose/_category_compositor.py

```diff
@@ -0,0 +1,260 @@
+
+    Parameters
+    ----------
+    forecasters : dict[sktime forecasters]
+        dict of forecasters with the key corresponding to categories generated
+        by the given transformer and the value corresponding to a sktime forecaster.
+
+    transformer : sktime transformer, default = ADICVTransformer()
+        A series-to-primitives sk-time transformer that generates a value
+        which can be used to quantify a choice of forecaster for the time series.
+
+    Raises
+    ------
+    AssertionError: If a valid transformer (an instance of BaseTransformer)
+    is not passed or if valid forecasters (instances of BaseForecaster) is not given.
+    """
+
+    _tags = {
+        "y_inner_mtype": "pd.Series",
+        "X_inner_mtype": "pd.Series",
+        "scitype:y": "univariate",
+        "ignores-exogeneous-X": True,
+        "requires-fh-in-fit": True,
+        "enforce_index_type": None,
+        "handles-missing-data": False,
+        "capability:insample": True,
+        "capability:pred_int": False,
+        "authors": ["shlok191"],
+        "maintainers": ["shlok191"],
+        "python_version": None,
+        "python_dependencies": [
+            "BaseForecaster",
+            "BaseTransformer" "ADICVTransformer",
+        ],
+    }
+
+    def __init__(self, forecasters, transformer=ADICVTransformer):
+        # saving arguments to object storage
+        self.transformer = transformer
+        self.forecasters = forecasters
+
+        super().__init__()
+
+        # validating passed arguments
+        assert isinstance(transformer, BaseTransformer)
+
+        for forecaster in forecasters.values():
+            assert isinstance(forecaster, BaseForecaster)
+
+        # All checks OK!
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit forecaster to training data.
+
+        private _fit containing the core logic, called from fit
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        y : Pd.Series
+            The target time series to which we fit the data.
+
+        fh : ForecastingHorizon | None, optional (default=None)
+            The forecasting horizon with the steps ahead to to predict.
+
+        X : Pd.Series | None, optional (default=None)
+            No exogenous variables are used for this.
+
+        Returns
+        -------
+        self : reference to self
+        """
+        # reset the state of the forecaster
+        self.reset()
+
+        # asserting validity of input
+        assert y is not None
+
+        # passing time series through the provided transformer!
+        self.category_ = self.transformer.fit_transform(y=y).iloc[0, 0]
+
+        # check if we have an available forecaster
+        assert (
+            self.category_ in self.forecasters
+        ), f"Forecaster not provided for given time series of type {self.category_}."
+
+        self.chosen_forecaster_ = self.forecasters[self.category_]
```

Comment:
makes sense, I wonder whether we can make it more robust by allowing the user to specify a `fallback_forecaster`, e.g., if the category is sth else. I would make it the third, optional argument, and use it as a default in a dict.get if it is not `None` - otherwise, raise a `ValueError` if no fallback is provided and the category is not found.

Reply:
I think that's definitely a good idea, I'll add it to the code :)